rct229/rules/section22/section22rule10.py

```diff
@@ -0,0 +1,148 @@
+        for chiller in find_all("$.chillers[*]", rmi_b):
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[chiller["cooling_loop"]] = ZERO.POWER
+            chw_loop_capacity_dict[chiller["cooling_loop"]] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return {
+            "loop_pump_dictionary": loop_pump_dictionary,
+            "chw_loop_capacity_dict": chw_loop_capacity_dict,
+            "primary_secondary_loop_dict": primary_secondary_loop_dict,
+        }
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+        return fluid_loop_b["id"] in chw_loop_capacity_dict.keys()
+
+    class PrimaryFluidLoopRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section22Rule10.PrimaryFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section22Rule10.PrimaryFluidLoopRule.SecondaryChildLoopRule(),
+                index_rmr="baseline",
+                list_path="$.child_loops[*]",
+            )
+
+        def create_data(self, context, data):
+            fluid_loop_b = context.baseline
+            chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+            if (
+                chw_loop_capacity_dict[fluid_loop_b["id"]]
+                < MAX_FIXED_SPEED_CHW_LOOP_COOLING_CAPACITY
+            ):
+                target_secondary_pump_type = PUMP_SPEED_CONTROL.FIXED_SPEED
+            else:
+                target_secondary_pump_type = PUMP_SPEED_CONTROL.VARIABLE_SPEED
+
+            return {**data, "target_secondary_pump_type": target_secondary_pump_type}
+
+        def list_filter(self, context_item, data):
+            child_loop_b = context_item.baseline
+            loop_pump_dictionary = data["loop_pump_dictionary"]
+
+            return child_loop_b["id"] in loop_pump_dictionary.keys()
+
+        class SecondaryChildLoopRule(RuleDefinitionBase):
+            def __init__(self):
+                super(
+                    Section22Rule10.PrimaryFluidLoopRule.SecondaryChildLoopRule, self
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                )
+
+            def get_calc_vals(self, context, data=None):
+                child_loop_b = context.baseline
+                loop_pump_dictionary = data["loop_pump_dictionary"]
+                secondary_pump_speed_control = getattr_(
+                    loop_pump_dictionary[child_loop_b["id"]],
+                    "speed_control",
+                    "speed_control",
+                )
+
+                return {"secondary_pump_speed_control": secondary_pump_speed_control}
```

Comment:
Should `target_secondary_pump_type` be included in `calc_vals` so it will be included in the output?